src/main/java/cloud/eppo/Configuration.java

```diff
@@ -0,0 +1,203 @@
+ *
+ * <p>Hint: when loading new Flag configuration values, set the current bandit models in the builder
+ * then check `requiresBanditModels()`.
+ */
+public class Configuration {
+  private static final Logger log = LoggerFactory.getLogger(Configuration.class);
+  private final Map<String, BanditReference> banditReferences;
+  private final Map<String, FlagConfig> flags;
+  private final Map<String, BanditParameters> bandits;
+  private final boolean isConfigObfuscated;
+
+  private Configuration(
+      Map<String, FlagConfig> flags,
+      Map<String, BanditReference> banditReferences,
+      Map<String, BanditParameters> bandits,
+      boolean isConfigObfuscated) {
+    this.flags = flags;
+    this.banditReferences = banditReferences;
+    this.bandits = bandits;
+    this.isConfigObfuscated = isConfigObfuscated;
+  }
+
+  public FlagConfig getFlag(String flagKey) {
+    String flagKeyForLookup = flagKey;
+    if (isConfigObfuscated()) {
+      flagKeyForLookup = getMD5Hex(flagKey);
+    }
+
+    if (flags == null) {
+      log.warn("Request for flag {} before flags have been loaded", flagKey);
+      return null;
+    } else if (flags.isEmpty()) {
+      log.warn("Request for flag {} with empty flags", flagKey);
+    }
+    return flags.get(flagKeyForLookup);
+  }
+
+  public String banditKeyForVariation(String flagKey, String variationValue) {
+    // Note: In practice this double loop should be quite quick as the number of bandits and bandit
+    // variations will be small. Should this ever change, we can optimize things.
+    for (Map.Entry<String, BanditReference> banditEntry : banditReferences.entrySet()) {
+      BanditReference banditReference = banditEntry.getValue();
+      for (BanditFlagVariation banditFlagVariation : banditReference.getFlagVariations()) {
+        if (banditFlagVariation.getFlagKey().equals(flagKey)
+            && banditFlagVariation.getVariationValue().equals(variationValue)) {
+          return banditEntry.getKey();
+        }
+      }
+    }
+    return null;
+  }
+
+  public BanditParameters getBanditParameters(String banditKey) {
+    return bandits.get(banditKey);
+  }
+
+  public boolean isConfigObfuscated() {
+    return isConfigObfuscated;
+  }
+
+  /**
+   * Builder to create the immutable config object.
+   *
+   * @see cloud.eppo.Configuration for usage.
+   */
+  public static class Builder {
+    private static final ObjectMapper mapper =
+        new ObjectMapper().registerModule(EppoModule.eppoModule());
+
+    private final boolean isConfigObfuscated;
+    private final Map<String, FlagConfig> flags;
+    private Map<String, BanditReference> banditReferences;
+    private Map<String, BanditParameters> bandits = Collections.emptyMap();
+
+    public static Configuration emptyConfig() {
+      return new Configuration(
+          Collections.emptyMap(), Collections.emptyMap(), Collections.emptyMap(), false);
+    }
+
+    public Builder(String flagJson) {
+      this(flagJson, false);
+    }
```

Comment:
minor: I would say there's no good default value for `isConfigObfuscated`, so I would rather remove this constructor.
```suggestion
```

Reply:
agreed